CONTRIBUTING.rst

```diff
@@ -59,61 +60,455 @@ If you are reporting a bug, please include:
+
+* Line 4: ``@pytest.mark.functions``
+
+This decorator is a `custom pytest mark <http://doc.pytest.org/en/latest/example/markers.html>`_. 
+You will often see it at the top of each test function following the convention of 
+``@pytest.mark.<module_name>``. 
+This mark allows us to restrict a test run to only run tests marked with `<module_name>`. 
+For example, we can run all the test with the ``pytest.mark.functions`` mark:
+
+.. code-block:: bash
+
+    # run `pytest -h` in your terminal to check all available options
+    $ pytest -v -m functions --cov
+
+Or conversely, we can run all the tests *except* the ``pytest.mark.functions`` ones:
+
+.. code-block:: bash
+
+    # run `pytest -h` in your terminal to check all available options
+    $ pytest -v -m "not functions" --cov
+
+* Line 5: ``dataframe`` injection
+
+Upon test function definition, the test ``dataframe`` fixture is injected.
+
+* Line 9-12: assertions for example-based tests
+
+After using the function in the test (line 6-8), 
+we use ``assert`` statements to carry out example-based tests using fixed inputs 
+and fixed, expected outputs.
+
+3. *How do we run the test?*
+
+* To run only your test:
+
+.. code-block:: bash
 
-This function should be implemented in ``functions.py``, and it should have a test
-accompanying it in ``tests/functions/test_<function_name_here>.py``.
+    # In `pyjanitor/tests/<module_name>`
+    $ pytest -v test_<function_name>.py --cov
 
-When writing a test, the minimum acceptable test is an "example-based test."
-Under ``tests/conf.py`` you will find a suite of example dataframes that can be
-imported and used in the test.
+* To run all the tests:
 
-If you are more experienced with testing, you can use `Hypothesis <https://hypothesis.readthedocs.io/en/latest/>`_ to
-automatically generate example dataframes. We provide a number of
-dataframe-generating strategies in ``janitor.testing_utils.strategies``.
+.. code-block:: bash
 
-We may go back-and-forth a few times on the docstring. The docstring is a very
-important part of developer documentation; therefore, we may want much more detail than
-you are used to providing. This is for maintenance purposes: Contributions from new contributors
-frequently end up being maintained by the maintainers, and hence we would err on the side of
-providing more contextual information than less, especially regarding design choices.
+    # Under `pyjanitor` top level directory (i.e., where `Makefile` is)
+    $ make test
 
-If you're wondering why we don't have to implement the method chaining
-portion, it's because we use the ``register_dataframe_method`` from ``pandas-flavor``,
-which registers the function as a pandas dataframe method.
+This is the basic structure of unit tests. 
+For your own tests, you can either use existing test data in ``conftest.py``, 
+or add your own test data to that file by following the same fixture pattern.
 
 Submit Feedback
-~~~~~~~~~~~~~~~
+-----------------
```

Comment:
Minor formatting suggestion: RST linters will usually make noise about the number of characters used to denote the header needing to match the number of characters in the heading text.

```suggestion
---------------
```